addons/website_sale_loyalty/views/website_sale_templates.xml

```diff
@@ -7,6 +7,17 @@
         <xpath expr="//t[@name='code_not_available']" position="replace"/>
     </template>
 
+    <template id="alert_messages" inherit_id="website_sale.cart">
+        <xpath expr="//div[@id='wrap']" position="before">
+            <t if="not promo_code_error">
+                <t t-set="promo_code_error" t-value="website_sale_order.get_promo_code_error()"/>
+            </t>
+            <t if="not promo_code_success">
+                <t t-set="promo_code_success" t-value="website_sale_order.get_promo_code_success_message()"/>
+            </t>
+        </xpath>
+    </template>
```

Comment:
Since we're in stable, can't we try finding a way to fix it without adding another template ? :thinking: 

First idea: give the two values as rendering values of the `website_sale.cart` template (in the `cart` route, with a clean hook in website_sale, extended in `website_sale_loyalty`)

Second idea: specify `delete=False` in all the calls to the methods, and manually clean the session content before returning the rendered template in the route.

Reply:
Thanks for the review 🙂  can you check the updated fix, something like this will be good? I tried to follow the first idea

Reply:
doesn't the solution I give work? I've not tested, but I had given a solution that would not add a template:

```diff
diff --git a/addons/website_sale_loyalty/views/website_sale_templates.xml b/addons/website_sale_loyalty/views/website_sale_templates.xml
index 94bb8ccb15c..a542ce4bf99 100644
--- a/addons/website_sale_loyalty/views/website_sale_templates.xml
+++ b/addons/website_sale_loyalty/views/website_sale_templates.xml
@@ -17,14 +17,16 @@
                                 Invalid or expired promo code.
                             </div>
                         </t>
-                        <t t-if="website_sale_order.get_promo_code_error(delete=False)">
+                        <t t-set="promo_code_error" t-value="website_sale_order.get_promo_code_error()" t-if="not promo_code_error"/>
+                        <t t-if="promo_code_error">
                             <div class="alert alert-danger text-start mt16" role="alert">
-                                <t t-esc="website_sale_order.get_promo_code_error()"/>
+                                <t t-esc="promo_code_error"/>
                             </div>
                         </t>
-                        <t t-if="website_sale_order and website_sale_order.get_promo_code_success_message(delete=False)">
+                        <t t-set="promo_code_success" t-value="website_sale_order.get_promo_code_success_message()" t-if="not promo_code_success"/>
+                        <t t-if="website_sale_order and promo_code_success">
                             <div class="alert alert-success text-start mt16" role="alert">
-                                You have successfully applied the following code: <strong t-esc="website_sale_order.get_promo_code_success_message()"/>
+                                You have successfully applied the following code: <strong t-esc="promo_code_success"/>
                             </div>
                         </t>
                         <t t-if="website_sale_order">
```

Reply:
Thanks for the suggestion :) I've tested but It have same behavior as we have now, we lose the messages on the second call